### PR TITLE
Added quizzes with questions and options

### PR DESCRIPTION
**Pull request by ishitakeshawani**
preview link: https://deploy-preview-3--memory-nomads.netlify.app/

### CODE REVIEW

File: src/pages/quiz-page/QuizPage.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            currentQuestionIndex={currentQuestionIndex}
+          />
+        </div>
+      </div>
+      <Footer />
+    </div>
+  );
+}
+
+const CurrentQuestion = ({ questionsList, currentQuestionIndex }) => {
```
Comment (rohanmathur91) [inline src/pages/quiz-page/QuizPage.js]: you can put this in new a file

File: src/pages/quiz-page/QuizPage.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  const { quizName } = useParams();
+  const { dispatch } = useQuiz();
+  const { quizState } = useQuiz();
```
Comment (rohanmathur91) [inline src/pages/quiz-page/QuizPage.js]: get both from anyone destructure

File: src/reducers/quizReducer.js [hunk 3]
```diff
@@ -6,6 +6,34 @@ export const quizReducer = (quizState, { type, payload }) => {
     case "SET_QUIZZES":
       return { ...quizState, quizList: payload };
 
+    case "SET_CURRENT_QUESTION_INDEX":
+      return { ...quizState, currentQuestionIndex: payload };
+
+    case "SET_SELECTED_OPTION":
+      return {
+        ...quizState,
+        quizList: quizState.quizList.map((quiz) =>
```
Comment (rohanmathur91) [inline src/reducers/quizReducer.js:17]: suggestion: put a comment explaining what is happening in this case

File: src/reducers/quizReducer.js [hunk 4]
```diff
@@ -6,6 +6,34 @@ export const quizReducer = (quizState, { type, payload }) => {
...
+                    : question
+                ),
+              }
+            : quiz
+        ),
+      };
+
     default:
       return quizState;
   }
```
Comment (rohanmathur91) [inline src/reducers/quizReducer.js]: instead of returning state throw an error

**Review by rohanmathur91 — APPROVED** [review]
Added a few comments, you can have a look, the rest looks good.

**Review by ranitasahaa1230 — APPROVED** [review]
Good use of contexts and reducers
Clean and readable code
Do work on the suggestions as suggested by Rohan
Rest looks good to merge

**Review by kumsomi — APPROVED** [review]
- everything is working smoothly and looks good.
- just a suggestion :- you can put an exit quiz button instead of showing home link on the quiz page. Because I'm able to switch to lhome page in between the ongoing quiz.

File: src/pages/quiz-page/QuizPage.js (hunk 1 above)
Comment (ishitakeshawani) [reply src/pages/quiz-page/QuizPage.js]: done

File: src/pages/quiz-page/QuizPage.js (hunk 2 above)
Comment (ishitakeshawani) [reply src/pages/quiz-page/QuizPage.js]: done

File: src/reducers/quizReducer.js (hunk 3 above)
Comment (ishitakeshawani) [reply src/reducers/quizReducer.js:17]: done

File: src/reducers/quizReducer.js (hunk 4 above)
Comment (ishitakeshawani) [reply src/reducers/quizReducer.js]: done

**Comment by ishitakeshawani** [comment]
> * everything is working smoothly and looks good.
> * just a suggestion :- you can put an exit quiz button instead of showing home link on the quiz page. Because I'm able to switch to lhome page in between the ongoing quiz.

Thank you for your feedback. I added quit button.

**Pull request merged**